### PR TITLE
Add HPO examples & results for tabular data with MultiModalPredictor

### PR DESCRIPTION
**Pull request by yiqings**
*Issue #, if available:*

*Description of changes:*

1. Add examples for HPO with MultiModalPredictor in tabular data. 
2. Add associated results.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/automm/tabular_dl/README.md [hunk 1]
```diff
@@ -1,28 +1,29 @@
-# Advanced Tabular DL models in AutoMM
+# Advanced models in autogluon.multimodal for tabular data
```
Comment (sxjscience) [inline examples/automm/tabular_dl/README.md:1]: I actually prefer the previous title.

File: examples/automm/tabular_dl/README.md [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
 
 To run the example: 
 
 ```python example_tabular.py --dataset_name ad --dataset_dir ./dataset --exp_dir ./result```
-   - `dataset_name` determines which dataset to run the experinments, refers to [Dataset Section](###1.2-Datasets).
+   - `dataset_name` determines which dataset to run the experinments, refers to [Dataset Section](###2.-Datasets).
    - `dataset_dir` is the path to the dataset(s). If the datasets do not present in this path, it will be automatically downloaded.
    - `exp_dir` is the output path to store the weights and loggings. 
-   - `seed` determines the random seed. Default is 0.
+   - `--gpu_id` specifies the GPU to use (optional).
```
Comment (sxjscience) [inline examples/automm/tabular_dl/README.md:13]: Let’s be consistent here. The other flags do not have “--“ so we should change it to ‘gpu_id’.

File: examples/automm/tabular_dl/README.md [hunk 3]
```diff
@@ -107,14 +108,42 @@ We can reproduce the following results by tuning `--embedding_arch` in `example_
...
+        "searcher": 'random',
+        "scheduler": 'FIFO',
+        "num_trials": 50,
+}
+```
+
+The results are as follws:
+Dataset (metric) | w/o HPO | w/ HPO | model_configs | results 
+----  | ----  | ----  | ----  | ----  
+ca (rmse) | 0.482 | 0.579 | [link](https://autogluon.s3.us-west-2.amazonaws.com/results/tabular/hpo/ca_hpo_config.yaml) | [link](https://autogluon.s3.us-west-2.amazonaws.com/results/tabular/hpo/ca_hpo_result.zip)
```
Comment (sxjscience) [inline examples/automm/tabular_dl/README.md:141]: we need to pay attention to the “ca” dataset. It is not reasonable why the performance decreased so much after the HPO.

File: examples/automm/tabular_dl/README.md [hunk 4]
```diff
@@ -107,14 +108,42 @@ We can reproduce the following results by tuning `--embedding_arch` in `example_
...
 
+### 6. Hyper-parameter optimization for FT_Transformer
+Set `--mode` in [`example_tabular.py`](./example_tabular.py) to `single_hpo` to run the HPO for FTTransformer. 
+The search spaces for FT_transformer are as follws:
+- "model.numerical_transformer.ffn_dropout": tune.uniform(0.0, 0.5), 
+- "model.numerical_transformer.attention_dropout": tune.uniform(0.0, 0.5),
+- "model.numerical_transformer.residual_dropout": tune.uniform(0.0, 0.2),
+- "optimization.learning_rate": tune.uniform(0.00001, 0.001),
+
+with the tuning kwargs as follws:
```
Comment (sxjscience) [inline examples/automm/tabular_dl/README.md:129]: follws --> follows

**Review by sxjscience — APPROVED** [review]
LGTM

File: multimodal/src/autogluon/multimodal/utils.py [hunk 5]
```diff
@@ -751,6 +751,7 @@ def create_model(
                 cls_token=True if len(names) == 1 else False,
                 embedding_arch=model_config.embedding_arch,
                 num_classes=num_classes,
+                ffn_d_hidden=model_config.ffn_d_hidden,
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/utils.py:754]: Try to use 

```python
ffn_d_hidden=OmegaConf.select(model_config, 'ffn_d_hidden', default=192)
```
